vassal-app/src/main/java/VASSAL/build/module/GameState.java

```diff
@@ -668,7 +683,70 @@ else if (result == JOptionPane.CANCEL_OPTION) { // "Don't Prompt Again" option.
     }
     catch (IOException e) {
       ReadErrorDialog.error(e, f);
+      return false;
     }
+
+    return true;
+  }
+
+  /**
+   * Accepts a saved file dropped onto the main window or a map, attempts to load it as a saved game.
+   * @param dtde DropTargetDropEvent from the drop() handler
+   */
+  public void dropFile(DropTargetDropEvent dtde) {
+    dtde.acceptDrop(DnDConstants.ACTION_COPY);
+    final Transferable transferable = dtde.getTransferable();
+    final DataFlavor[] flavors = transferable.getTransferDataFlavors();
+    for (final DataFlavor flavor : flavors) {
+
+      // If the drop item is a Discord file link (or a text URL), we attempt to open the connection and Load That Shit Right Off The Internet
+      if (flavor.isFlavorTextType()) {
+        try {
+          final String text = transferable.getTransferData(flavor).toString();
+
+          try {
+            final URL url = new URL(text);
+            final URLConnection uc = url.openConnection();
+            final InputStream is = new BufferedInputStream(uc.getInputStream());
+
+            try {
+              loadGameInForeground(text, is);
+            }
+            catch (IOException e) {
+              ReadErrorDialog.error(e, text);
+            }
+            break; // Once we have a successful load, nothing else.
```

Comment:
Is this because you can't tell beforehand which flavor will be useful?

Reply:
The "transferrables" that we get seem to come in multiple "flavors" and indeed as far as I can tell the same individual thing will sometimes present itself "in several different flavors". 

And of course all manner of shit _might_ get dragged here (totally inappropriate things that aren't files -- here I dragged a paragraph I copied from my Word document) that come in various comprehensible/incomprehensible flavors. 

So what we're doing here is:
(a) If we got some kind of "text paste", see if it has a URL to a loadable game file. The Discord links that you drag in from the chat rooms come with a lot of extra cruft but they have a URL in them you can dig out. 
(b) If we got a "file" paste then see if it's a loadable game file.